server/lib/routes/get-verify-email.js

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const got = require('got');
+const logger = require('mozlog')('server.get-verify-email');
+const config = require('../configuration');
+const joi = require('joi');
+const validation = require('../validation');
+
+const fxaAccountUrl = config.get('fxaccount_url');
+const STRING_TYPE = validation.TYPES.STRING;
+
+const BODY_SCHEMA = {
+  'code': STRING_TYPE.alphanum().min(32).max(32).required(),
+  'uid': STRING_TYPE.alphanum().min(32).max(32).required()
+};
+
+module.exports = function () {
+
+  return {
+    method: 'get',
+    path: '/verify_email/',
+    process: function (req, res, next) {
+      req.url = '/';
+
+      const data = {
+        code: req.query.code,
+        uid: req.query.uid
+      };
+
+      joi.validate(data, BODY_SCHEMA, function (err, value) {
+        if (err) {
+          logger.error(err);
```

Comment:
@divyabiyani in a case of an error it seems like `next();` will never be called and the request will hang forever